staging/src/k8s.io/apimachinery/pkg/apimachinery/announced/group_factory.go

```diff
@@ -89,6 +89,17 @@ func (gmf *GroupMetaFactory) Announce(groupFactoryRegistry APIGroupFactoryRegist
 		panic(err)
 	}
 	return gmf
+
+}
+
+// AnnounceIfNotAnnounced is the same as Announce, except that it returns nil
```

Comment:
I'll try to eliminate the double registration if that's not too messy. Client have actually defined its own copy of scheme/registry/APIGroupFactoryRegistry in client-go/kubernetes/scheme, but i guess they are used in mix with api.Scheme.